src/style-spec/function/definitions/curve.js

```diff
@@ -31,34 +26,23 @@ class Curve implements Expression {
     input: Expression;
     stops: Stops;
 
-    interpolationCache: InterpolationCache;
-
     constructor(key: string, type: Type, interpolation: InterpolationType, input: Expression, stops: Stops) {
         this.key = key;
         this.type = type;
         this.interpolation = interpolation;
         this.input = input;
         this.stops = stops;
-
-        this.interpolationCache = {pow: {}, unitBezier: {}};
     }
 
-    static interpolationFactor(interpolation: InterpolationType, input: number, lower: number, upper: number, cache: InterpolationCache) {
+    static interpolationFactor(interpolation: InterpolationType, input: number, lower: number, upper: number) {
         let t = 0;
         if (interpolation.name === 'exponential') {
-            t = exponentialInterpolation(input, interpolation.base, lower, upper, cache.pow);
+            t = exponentialInterpolation(input, interpolation.base, lower, upper);
         } else if (interpolation.name === 'linear') {
-            t = exponentialInterpolation(input, 1, lower, upper, cache.pow);
+            t = exponentialInterpolation(input, 1, lower, upper);
         } else if (interpolation.name === 'cubic-bezier') {
-            const key = interpolation.controlPoints.join(',');
-            let ub = cache.unitBezier[key];
-            if (!ub) {
-                ub = new UnitBezier(...interpolation.controlPoints);
-                if (cache.unitBezier) {
-                    cache.unitBezier[key] = ub;
-                }
-            }
-            t = ub.solve(exponentialInterpolation(input, 1, lower, upper, cache.pow));
+            const ub = new UnitBezier(...interpolation.controlPoints);
```

Comment:
Let's change this to explicit `p[0], p[1], p[2], p[3]`. I just realized that this can be a huge hit because Buble [compiles this](http://buble.surge.sh/#const%20ub%20%3D%20new%20UnitBezier(...interpolation.controlPoints)%3B) to the following atrocity:

```js
var ub = new (Function.prototype.bind.apply( UnitBezier, [ null ].concat( interpolation.controlPoints) ));
```

Reply:
oof good call